### ISSUE TITLE
No password option when connecting to host

### ISSUE THREAD

**Issue by BorntraegerMarc** [eclipse/paho.mqtt.javascript#55]
So in the example:
http://www.eclipse.org/paho/clients/js/utility/index.html
You can specify a host, port and username but no password! I also did not find an option for that in the lib... For my cloud under https://www.cloudmqtt.com/ there is a password needed for the client.

Is specifying a password not possible?

cheers,
marc

**Comment by jpwsutton** [comment]
There is definitely a password option (in screenshot below). You can see it being used in the utility script here: https://github.com/eclipse/paho.mqtt.javascript/blob/develop/utility/utility.js#L149

![screen shot 2016-03-25 at 13 45 35](https://cloud.githubusercontent.com/assets/1068763/14045197/ed5dab5e-f28f-11e5-8248-81ee7c1a19f1.png)

The docs for the connectOptions are also here: http://www.eclipse.org/paho/files/jsdoc/symbols/Paho.MQTT.Client.html#connect
All you need to do is add a password field to your connectOptions object :)

**Comment by BorntraegerMarc** [comment]
Hi @jpwsutton 

Thanks for the quick reply :)
As I understand it:
The username and password is used for the actual device, that connects to the server, am I right?

Now it's a little bit different with this cloud. I have some information for the connection to the cloud:
![image](https://cloud.githubusercontent.com/assets/7549295/14045617/d8d244b8-f29b-11e5-8f8d-4beb270b7c55.png)

And then there's the information for the specific device:
![image](https://cloud.githubusercontent.com/assets/7549295/14045619/e2c24f86-f29b-11e5-95e1-3c1f3ea474ec.png)

Please note, that the username and password for each screenshot is different. So is the by you specified password for both connections? Or how is the lib handling this situation?

**Comment by BorntraegerMarc** [comment]
Also, when I try to connect to the host, then I'm receiving the following error:
Connecting to Server: Hostname:  m21.cloudmqtt.com . Port:  11486 . Path:  /lightState . Client ID:  web

mqttws31.js:979 WebSocket connection to 'ws://m21.cloudmqtt.com:11486/lightState' failed: Error during WebSocket handshake: net::ERR_CONNECTION_RESET`

**Comment by jpwsutton** [comment]
So as I understand it (From a quick look on the cloudmqtt docs) every device should use the same username and password unless you use their ACL functionality  in which case every device should have a different username and password that you set up from the control panel or the HTTP api (https://www.cloudmqtt.com/docs-api.html).

So if you're not using ACLS, use the main username and password for the instance (the ones blanked out in your screenshot), if you are using ACLS, use the username and password shown on that devices page.

In terms of your second question, it looks like you are using the wrong port for WebSockets. In the screenshot, it shows the WebSockets port as 31486.

**Comment by BorntraegerMarc** [comment]
Alright, thanks @jpwsutton for the info! So, I tried out some things and gained new info:

1. The ACL is device specific and as far as I know, every device **needs** a separate username + password to read and write the data. Sure they could all go with the same credentials, but that wouldn't be very secure, would it? ;)
2. I tried out the websockets port for the web application and it worked! Thank you very much :) Be sure to activate the 'SSL' option if you're connecting, else it won't work...

So to sum it up:

- Host on JS is the Server on CloudMQTT
- Port on JS is the Websockets Port (TLS only) on CloudMQTT
- Client ID on JS is the User on CloudMQTT
- Username and pw on JS is the specific ACL credentials on CloudMQTT
- Be sure to activate the 'SSL' option

Thanks! Going to close the issue...
Cheers!
